modules/siddhi-core/src/main/java/org/wso2/siddhi/core/util/parser/AggregationParser.java

```diff
@@ -293,6 +294,15 @@ public static AggregationRuntime parse(AggregationDefinition aggregationDefiniti
                         SiddhiConstants.METRIC_INFIX_WINDOWS, SiddhiConstants.METRIC_TYPE_INSERT);
 
             }
+            //initialize incremental data purging
```

Comment:
Handling purging should be handled by the aggregation runtime, and purging should only be started when siddhi app start is called.